pkg/syncer/resourcesync/controller.go

```diff
@@ -0,0 +1,348 @@
+			cancel()
+			delete(c.syncerMap, gvr)
+		}
+	}
+}
+
+func (c *Controller) startSyncer(ctx context.Context, gvr schema.GroupVersionResource, syncTarget *workloadv1alpha1.SyncTarget) error {
+	logger := klog.FromContext(ctx)
+
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	if _, ok := c.syncerMap[gvr]; ok {
+		return nil
+	}
+
+	if len(syncTarget.Status.VirtualWorkspaces) == 0 {
+		return nil
+	}
+
+	syncerVirtualWorkspaceURL := syncTarget.Status.VirtualWorkspaces[0].URL
+
+	upstreamURL, err := url.Parse(syncerVirtualWorkspaceURL)
+	if err != nil {
+		return err
+	}
+
+	// Check whether we're in the Advanced Scheduling feature-gated mode.
+	advancedSchedulingEnabled := false
+	if syncTarget.GetAnnotations()[AdvancedSchedulingFeatureAnnotation] == "true" {
+		logger.WithValues("SyncTarget", c.syncTargetName).Info("Advanced Scheduling feature is enabled")
+		advancedSchedulingEnabled = true
+	}
+
+	syncTargetKey := workloadv1alpha1.ToSyncTargetKey(c.syncTargetWorkspace, c.syncTargetName)
+
+	upstreamInformers := dynamicinformer.NewFilteredDynamicSharedInformerFactoryWithOptions(c.upstreamDynamicClusterClient.Cluster(logicalcluster.Wildcard), metav1.NamespaceAll, func(o *metav1.ListOptions) {
+		o.LabelSelector = workloadv1alpha1.ClusterResourceStateLabelPrefix + syncTargetKey + "=" + string(workloadv1alpha1.ResourceStateSync)
+	})
+	downstreamInformers := dynamicinformer.NewFilteredDynamicSharedInformerFactoryWithOptions(c.downstreamDynamicClient, metav1.NamespaceAll, func(o *metav1.ListOptions) {
+		o.LabelSelector = workloadv1alpha1.InternalDownstreamClusterLabel + "=" + syncTargetKey
+	}, cache.WithResyncPeriod(resyncPeriod), cache.WithKeyFunction(keyfunctions.DeletionHandlingMetaNamespaceKeyFunc))
+
+	specSyncer, err := spec.NewSpecSyncer(gvr, c.syncTargetWorkspace, c.syncTargetName, syncTargetKey, upstreamURL, advancedSchedulingEnabled,
+		c.upstreamDynamicClusterClient, c.downstreamDynamicClient, upstreamInformers.ForResource(gvr), downstreamInformers.ForResource(gvr), c.downstreamNSInformer, c.upstreamSecretInformer, c.syncTargetUID)
+	if err != nil {
+		return err
+	}
+
+	statusSyncer, err := status.NewStatusSyncer(gvr, c.syncTargetWorkspace, c.syncTargetName, syncTargetKey, advancedSchedulingEnabled,
+		c.upstreamDynamicClusterClient, c.downstreamDynamicClient, upstreamInformers.ForResource(gvr), downstreamInformers.ForResource(gvr), c.downstreamNSInformer, c.syncTargetUID)
+	if err != nil {
+		return err
+	}
+
+	logger.WithValues("gvr", gvr.String()).V(2).Info("Start syncer for gvr")
+	syncerCtx, cancel := context.WithCancel(ctx)
+
+	go specSyncer.Start(syncerCtx, c.numSyncerThreads)
+	go statusSyncer.Start(syncerCtx, c.numSyncerThreads)
+	go downstreamInformers.Start(syncerCtx.Done())
+	go upstreamInformers.Start(syncerCtx.Done())
+
+	c.syncerMap[gvr] = cancel
+	return nil
+}
+
+func getAllGVRs(synctarget *workloadv1alpha1.SyncTarget) (map[schema.GroupVersionResource]bool, error) {
+	// add configmap and secrets by default
```

Comment:
Why? We have a list of built-in types that's more than just this, and regardless don't we have e.g. the `kubernetes` export with these bits?

Reply:
This is from the old code :) https://github.com/kcp-dev/kcp/blob/main/pkg/syncer/syncer.go#L292. I think the reason is we do not sync cluster scoped and we do not sync sa. The kubernetes export does not include configmaps and secrets.